.github/workflows/build.yml

```diff
@@ -0,0 +1,39 @@
+name: build-and-test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: "Setup Rust toolchain"
+      uses: actions-rs/toolchain@v1
+      with:
+        # This has to be kept in sync with the charon/rust-toolchain file,
+        # except if the toolchain file is added to the root.
+        toolchain: nightly-2022-01-29
+        components: rustc-dev, llvm-tools-preview
+    - name: "build rs"
+      run: make
+    - name: "test rs"
+      run: make test
+    - name: "Setup ocaml 4.12.1"
+      uses: ocaml/setup-ocaml@v2
+      with:
+        ocaml-compiler: 4.12.x
```

Comment:
why not the latest?

Reply:
OCaml maintains backward compatibility, so as a rule of thumbs we might as well use an older version.
I chose 4.12 because that's what [Aeneas says it uses](https://github.com/AeneasVerif/aeneas#installation--build), but  anything is fine there